pkg/registry/blobs.go

```diff
@@ -31,6 +32,7 @@ type blobs struct {
 	contents map[string][]byte
 	// Each upload gets a unique id that writes occur to until finalized.
 	uploads map[string][]byte
+	lock    sync.Mutex
```

Comment:
https://golang.org/pkg/sync/#RWMutex instead?

Reply:
It's simpler to not use it - If people actually run into performance issues that we can flip over.